uplink/auth.py

```diff
@@ -43,9 +50,63 @@ def __call__(self, request_builder):
         request_builder.info["headers"]["Proxy-Authorization"] = self._auth_str
 
 
-class BearerToken(object):
+class ApiTokenParam(object):
+    """
+    Authorizes requests using a token or key in a query parameter.
+    Users should subclass this class to define which parameter is the token parameter.
+    """
+    param = None
+
     def __init__(self, token):
-        self._auth_str = "Bearer %s" % token
+        self._param_value = token
 
     def __call__(self, request_builder):
-        request_builder.info["headers"]["Authorization"] = self._auth_str
+        if not self.param:
+            raise NotImplementedError
+        request_builder.info["params"][self.param] = self._param_value
+
+
+class ApiTokenHeader(object):
+    """
+    Authorizes requests using a token or key in a header.
+    Users should subclass this class to define which header is the token header.
+    The subclass may also, optionally, define a token prefix (such as in BearerToken)
+    """
```

Comment:
Similar to ``ApiTokenParam``, I think we could let users pass the header as a constructor value instead of requiring them to subclass here.

Reply:
See my comment on `ApiTokenParam`

Reply:
subclass that specifies the param as a class param
```
class BearerToken(ApiTokenHeader):
    _header = "Authorization"
￼    _prefix = "Bearer"
```

subclass that uses the constructor
```
class BearerToken(ApiTokenHeader):
    def __init__(self, token):
        super(BearerToken, self).__init__(header='Authorization', prefix='Bearer', token=token)
```